examples/nucleo_f401re/adc_dma/main.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2023, Zühlke Engineering (Austria) GmbH
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+// This Example uses adc to sample 2 channels in scan mode 50 times every 0.5s and store the value
+// in a buffer via dma. The adc is set to be triggered by the Timer1 compare event and then to
+// sample both channels using scan mode. The compare event is triggered for a defined number of
+// times by configuring the Timer1 to use one pulse mode with an repetition count. For more
+// information on the timer please refer to the timer register count example. In this configuration,
+// the adc sampling process is started by the cpu but handled completely by peripherals clearing CPU
+// time for other tasks.
+
+#include <modm/architecture/interface/interrupt.hpp>
+#include <modm/board.hpp>
+#include <modm/platform.hpp>
+
+#include "adc_dma.hpp"
+#include "timer_handler.hpp"
+
+using namespace Board;
+using namespace modm::platform;
+using Adc1Dma = AdcDma<Adc1, Dma2::Channel0>;
+
+std::array<uint16_t, 100> adc_results;
+volatile bool dma_completed = false;
+
+void
+completedCallback()
+{
+	LedD13::toggle();
+	dma_completed = true;
+}
+
+int
+main()
+{
+	Board::initialize();
+	Adc1::initialize<SystemClock>();
+	Dma2::enable();
+
+	// Use the logging streams to print some messages.
+	// Change MODM_LOG_LEVEL above to enable or disable these messages
+	MODM_LOG_DEBUG << "debug" << modm::endl;
+	MODM_LOG_INFO << "info" << modm::endl;
+	MODM_LOG_WARNING << "warning" << modm::endl;
+	MODM_LOG_ERROR << "error" << modm::endl;
+
+	LedD13::setOutput();
+	LedD13::reset();
+	advancedTimerConfig<Timer1>(adc_results.size() / 2 - 1);
```

Comment:
One last question, why is the repetition counter set to `adc_results.size() / 2 - 1` here now that the timer interrupt handler is gone? It should only have an effect on the generation of update events. The ADC is triggered by the channel 1 Capture Compare reference signal. I can't see how the repetition counter affects ADC sampling here.

Reply:
It's because I want the PWM to generate only `adc_results.size()/2` pulses. For that you can set the Timer in onePulsemode and set a value in the rcr register. As said in the second answer of this entry. 
https://stackoverflow.com/questions/62217872/generate-a-fixed-number-of-pulses-on-the-stm32f4-pwm